packages/core/src/App/Components/Layout/Header/platform-switcher.jsx

```diff
@@ -17,6 +17,10 @@ class PlatformSwitcher extends React.PureComponent {
         this.state = { is_open: false };
     }
 
+    componentDidUpdate() {
+        this.props.filterNotificationMessages();
+    }
+
```

Comment:
Rather than using `componentDidUpdate` we could use a Mobx `reaction()` that will fire when `common_store.app_routing_history` was changed. This reaction would take care of serving the correct notifications as well as disposing `is_disposable` notifications.